src/main/java/UseCase/ViewProfileInputManager.java

```diff
@@ -0,0 +1,33 @@
+package UseCase;
+
+import Entities.UserAccount;
+import Entities.UserDatabase;
+import InterfaceAdapters.ViewProfilePresenter;
+
+import java.util.ArrayList;
+
+public class ViewProfileInputManager implements ViewProfileInputBoundary {
+
+    public ViewProfileInputManager() {}
+
+    public ArrayList searchInfo(UserDatabase database, String username) {
+        DatabaseManager databaseManager = DatabaseManager.getDatabaseManager();
+        UserAccount account = databaseManager.retrieve_user_account(username);
+
+        ArrayList info = new ArrayList<>(11);
+        info.add(account.get_username());
+        info.add(account.get_full_name());
+        info.add(account.get_age());
+        info.add(account.get_pronouns());
+        info.add(account.get_country());
```

Comment:
Is there a more efficient way to do this in terms of typing it out? Your work is good but it can be hard to alter if you need to change it in the future.